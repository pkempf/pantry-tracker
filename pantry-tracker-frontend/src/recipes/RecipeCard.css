.CompanyCard {
  color: black;
  text-decoration: none;
}

.CompanyCard:hover {
  text-decoration: none;
}

.CompanyCard:hover > div {
  box-shadow: rgba(0, 0, 0, 0.35) 0px 5px 15px;
}
