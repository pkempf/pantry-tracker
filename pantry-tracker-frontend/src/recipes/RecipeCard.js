import React from "react";
import Card from "react-bootstrap/Card";
import { Link } from "react-router-dom";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import Button from "react-bootstrap/Button";
import "./RecipeCard.css";

const RecipeCard = ({ recipe, add, remove }) => {
  return (
    <Card className="mb-2 RecipeCard">
      <Card.Body>
        <Link to={`/recipes/${recipe.id}`}>
          <Card.Title>
            <b>{recipe.name}</b>
          </Card.Title>
        </Link>
        <Card.Subtitle className="mb-2 text-muted">
          {recipe.category}
        </Card.Subtitle>
        <Row>
          <Col sm={9}>
            <span className="font-weight-light">{recipe.area}</span>
          </Col>
          <Col sm={3}>
            <div className="d-flex justify-content-end m-0 p-0">
              {recipe.isFavorite ? (
                <Button variant="text" onClick={remove}>
                  <i className="fa fa-heart" aria-label="Remove favorite"></i>
                </Button>
              ) : (
                <Button onClick={add} variant="text">
                  <i className="fa fa-heart-o" aria-label="Add favorite"></i>
                </Button>
              )}
            </div>
          </Col>
        </Row>
      </Card.Body>
    </Card>
  );
};

export default RecipeCard;
