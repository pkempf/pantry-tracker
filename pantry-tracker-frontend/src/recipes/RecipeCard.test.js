import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import RecipeCard from "./RecipeCard";

it("renders without crashing", () => {
  render(
    <MemoryRouter>
      <RecipeCard recipe={{}} />
    </MemoryRouter>
  );
});

it("matches the snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <RecipeCard recipe={{}} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
