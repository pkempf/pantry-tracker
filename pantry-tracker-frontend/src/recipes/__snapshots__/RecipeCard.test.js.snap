// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-2 RecipeCard card"
  >
    <div
      class="card-body"
    >
      <a
        href="/recipes/undefined"
      >
        <div
          class="card-title h5"
        >
          <b />
        </div>
      </a>
      <div
        class="mb-2 text-muted card-subtitle h6"
      />
      <div
        class="row"
      >
        <div
          class="col-sm-9"
        >
          <span
            class="font-weight-light"
          />
        </div>
        <div
          class="col-sm-3"
        >
          <div
            class="d-flex justify-content-end m-0 p-0"
          >
            <button
              class="btn btn-text"
              type="button"
            >
              <i
                aria-label="Add favorite"
                class="fa fa-heart-o"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
